utils/S3UtilsMongoClient.js

```diff
@@ -44,11 +45,15 @@ class S3UtilsMongoClient extends MongoClientInterface {
 
             const locationConfig = getLocationConfig(log);
 
-            const usersBucketCreationDatesArray = await this.getCollection(USERSBUCKET).find({}, {
+            const cursorUsersBucketCreationDates = await this.getCollection(USERSBUCKET).find({}, {
                 projection: {
                     'value.creationDate': 1,
                 },
-            }).toArray();
+            });
+
+            const usersBucketCreationDatesArray = await cursorUsersBucketCreationDates.toArray();
+
+            await cursorUsersBucketCreationDates.close();
```

Comment:
shouldn't we introduce a (small) `finally` block to close this cursor, to ensure we close the cursor in case `toArray()` fails with an exception?

```suggestion
	const usersBucketCreationDatesArray = await async() {
	   try {
              return await cursorUsersBucketCreationDates.toArray();
           } finally {
	      await cursorUsersBucketCreationDates.close();
	   }
	}();
```

(may be cleaner to extract this and `cursorUsersBucketCreationDates` creation to a separate method, though)

Reply:
Done in https://github.com/scality/s3utils/pull/310/commits/5017e66391393dbb5a59114897abf68d59efdb21 , with a new function (unit tested)